includes/results.inc

```diff
@@ -939,30 +939,49 @@ class IslandoraSolrFacets {
     $soft_limit = self::$soft_limit;
     $buckets = array();
     $replace_bucket = (isset($this->settings['solr_field_settings']['pid_object_label']) && $this->settings['solr_field_settings']['pid_object_label'] ? TRUE : FALSE);
+    $pid_mapper = function ($result) {
+      $pid = str_replace('info:fedora/', '', $result['bucket']);
+      return $pid;
+    };
+    if ($replace_bucket) {
+      $valid_pids = array_filter(array_map($pid_mapper, $results), 'islandora_is_valid_pid');
+      $mapping = array();
+      foreach ($valid_pids as $valid_pid) {
+        $mapping[$valid_pid] = "\"{$valid_pid}\"";
+      }
+      $qp = new IslandoraSolrQueryProcessor();
+      $qp->buildQuery(format_string('PID:(!pids)', array(
+        '!pids' => implode(' OR ', $mapping),
+      )));
+      $label_field = variable_get('islandora_solr_object_label_field', 'fgs_label_s');
+      $qp->solrParams['facet'] = 'false';
+      $qp->solrParams['fl'] = "PID, $label_field";
+      $qp->solrLimit = count($mapping);
```

Comment:
Not sure [about this](https://github.com/Islandora/islandora_solr_search/pull/322/files#diff-17a5945b1aa412dec3b39453ad29425fR959), but, if `$qp->solrLimit = count($mapping);` then in the case of no mapping at all, we should simply avoid the logic and don`t even query at all instead of putting a limit of `0` right?